### ISSUE TITLE
List subscriptions with user's role

### ISSUE THREAD

**Issue by jroslaniec** [Azure/azure-cli#6413]
We are using azure-cli for our automation scripts, and we would like to get a list of all subscriptions that user is an owner of (or an administrator).
 
We couldn't find any easy way to do it, so we would do it as follow:

1. `az account list` and retrieved an email from `user.name`

2. Retrieve user's objectId by `az ad user list --query "[?mail=='<EMAIL>']"`

3. And finally execute `az role assignment list --query "[?principalId=='<OBJECT_ID>']"` to get a role.

The main issue is that in some cases in the 2. step, not every user has a mail assigned to it (why?).
Also, all these steps are a bit cumbersome, especially if a user has lots of subscriptions. I hope there is a better way.

To summarise, is there a way to retrieve current user's objectId or list all subscriptions with user's role(s) in it?

**Comment by yugangw-msft** [comment]
@jroslaniec, to understand your context, one question here, `az role assignment list` already outputs the `principalName` which you can use to match with the `user.name`. Any reason you don't use that?
